config/config_test.go

```diff
@@ -0,0 +1,42 @@
+package config
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestConfig(t *testing.T) {
+	os.Clearenv()
+	var err error
+	var configuration *Config
+
+	Convey("Given an environment with no environment variables set", t, func() {
+		Convey("Then cfg should be nil", func() {
+			So(cfg, ShouldBeNil)
+		})
+
+		Convey("When the config values are retrieved", func() {
+
+			Convey("Then there should be no error returned, and values are as expected", func() {
+				configuration, err = Get() // This Get() is only called once, when inside this function
+				So(err, ShouldBeNil)
+				So(configuration, ShouldResemble, &Config{
+					BindAddr:                   "localhost:25700",
+					GracefulShutdownTimeout:    5 * time.Second,
```

Comment:
Graceful shutdown time = 5 , seems very short. We have been changing this to 20s for many of our apps

Reply:
And if we change it to 20, we need to update in few places

Reply:
I've changed it wherever I've found it - please let me know if you spot any that I've missed